core-bundle/contao/widgets/RowWizard.php

```diff
@@ -0,0 +1,278 @@
+				{
+					continue;
+				}
+
+				$widget->validate();
+
+				if ($widget->hasErrors())
+				{
+					$this->objDca->noReload = true;
+				}
+				else
+				{
+					$varValue[$i][$key] = $widget->value;
+				}
+			}
+		}
+
+		if ($this->hasErrors())
+		{
+			$this->class = 'error';
+		}
+
+		$this->varValue = $varValue;
+	}
+
+	public function generate(): string
+	{
+		// Make sure there is at least an empty array
+		if (!\is_array($this->varValue) || array() === $this->varValue)
+		{
+			$this->varValue = array(array(''));
+		}
+
+		// Populate the rows if the initial count has not been reached
+		if (null !== $this->min)
+		{
+			$rowCount = \count($this->varValue);
+
+			while ($rowCount < $this->min)
+			{
+				$this->varValue[] = array('');
+				++$rowCount;
+			}
+		}
+
+		$header = $rows = array();
+
+		for ($i = 0, $c = \count($this->varValue); $i < $c; ++$i)
+		{
+			$columns = array();
+			$header = array();
+
+			foreach ($this->arrFields as $key => $options)
+			{
+				if (\is_array($options['input_field_callback'] ?? null))
+				{
+					$header[] = array();
+					$columns[] = System::importStatic($options['input_field_callback'][0])->{$options['input_field_callback'][1]}($this->objDca);
+					continue;
+				}
+
+				if (\is_callable($options['input_field_callback'] ?? null))
+				{
+					$header[] = array();
+					$columns[] = $options['input_field_callback']($this->objDca);
+					continue;
+				}
+
+				$widget = $this->prepareWidget($key, $this->varValue[$i][$key] ?? null, $options, $i);
+
+				if (null !== $widget)
+				{
+					if ('be_widget' === $widget->template)
+					{
+						$header[] = array('label' => $widget->label ?? '', 'mandatory' => $widget->mandatory);
+						$widget->label = null;
+						$widget->template = $this->strTemplate;
+					}
+					else
+					{
+						$header[] = array();
+					}
+
+					$columns[] = $widget->generateWithError(true);
+				}
+			}
+
+			$rows[$i] = array(
+				'columns' => $columns,
+				'controls' => array(
+					'enable' => $this->varValue[$i]['enable'] ?? false,
+					'edit' => ($this->varValue[$i]['id'] ?? 0) > 0,
+				),
+			);
+		}
+
+		return System::getContainer()->get('twig')->render('@Contao/backend/widget/row_wizard.html.twig', array(
+			'id' => $this->strId,
+			'style' => $this->style,
+			'header' => $header,
+			'rows' => $rows,
+			'min_rows' => $this->min,
+			'max_rows' => $this->max,
+			'sortable' => $this->sortable,
+			'actions' => $this->actions,
+		));
+	}
+
+	public static function getAttributesFromDca($arrData, $strName, $varValue = null, $strField = '', $strTable = '', $objDca = null): array
+	{
+		$attributes = parent::getAttributesFromDca($arrData, $strName, $varValue, $strField, $strTable, $objDca);
+
+		if (isset($arrData['fields']) && !isset($attributes['fields']))
+		{
+			$attributes['fields'] = $arrData['fields'];
+		}
+
+		return $attributes;
+	}
+
+	private function prepareWidget(string $type, mixed $value, array $options, int $increment): Widget|null
```

Comment:
```suggestion
	private function prepareWidget(string $name, mixed $value, array $options, int $increment): Widget|null
```
Should we rename this to `$name` or `$key`?